combat-trainer.lic

```diff
@@ -295,73 +295,74 @@ class LootProcess
-    game_state.mob_died = true
-    healing = game_state.casting_consume || game_state.prepare_consume
-    if @necro && @last_ritual.nil?
+    if @last_ritual.nil?
       if @heal
         game_state.wounds = check_health
         echo "Severity to Wounds: #{game_state.wounds}" if $debug_mode_ct
         unless game_state.wounds.empty?
-          game_state.prepare_consume = healing = check_thanatology(DRRoom.dead_npcs.first, 'preserve', game_state) && check_thanatology(DRRoom.dead_npcs.first, 'consume', game_state)
+          do_ritual(DRRoom.dead_npcs.first, 'preserve', game_state)
+          do_ritual(DRRoom.dead_npcs.first, 'consume', game_state)
         end
       end
-      if @ritual_type == 'harvest'
-        check_thanatology(DRRoom.dead_npcs.first, 'preserve', game_state)
-      end
-      check_thanatology(DRRoom.dead_npcs.first, @ritual_type, game_state) unless healing
-      if ['preserve'].include?(@ritual_type) && !healing
-        arrange_mob(DRRoom.dead_npcs.first, game_state)
-        check_skinning(DRRoom.dead_npcs.first, game_state)
-      end
-    elsif @skin && !%w(consume harvest dissect).include?(@last_ritual)
-      arrange_mob(DRRoom.dead_npcs.first, game_state)
-      check_skinning(DRRoom.dead_npcs.first, game_state)
-    end
-    unless healing
-      while 'and get ready to search it' == bput('loot', 'You search', 'I could not find what you were referring to', 'and get ready to search it')
-        pause
-        waitrt?
+
+      if @ritual_type == 'harvest' && ! game_state.prepare_consume
+        do_ritual(DRRoom.dead_npcs.first, 'preserve', game_state)
+        do_ritual(DRRoom.dead_npcs.first, @ritual_type, game_state)
+      elsif @ritual_type && ! game_state.prepare_consume
+        do_ritual(DRRoom.dead_npcs.first, @ritual_type, game_state)
       end
-      @last_ritual = nil
     end
-    @loot_timer = Time.now
   end
 
-  def check_thanatology(mob_noun, ritual, game_state)
-    return false if game_state.construct?(mob_noun)
+  def do_ritual(mob_noun, ritual, game_state)
+    return unless DRStats.necromancer?
+    return if game_state.construct?(mob_noun)
 
     perform_message = "perform #{ritual} on #{mob_noun}"
-    case bput(perform_message, 'Rituals do not work upon constructs', 'Rituals do not work upon the undead', 'This ritual may only be performed on a corpse', 'precise motions with your ritual knife', 'corpse to make one long', 'carefully position the corpse', 'corpse and make a few quick precise cuts', 'roundtime')
-    when 'corpse to make one long', 'carefully position the corpse', 'corpse and make a few quick precise cuts', 'Your cuts follow the vital lines of the corpse'
+    case bput(perform_message, 'Rituals do not work upon constructs', 'Rituals do not work upon the undead', 'This ritual may only be performed on a corpse', 'precise motions with your ritual knife', 'corpse to make one long', 'carefully position the corpse', 'roundtime', 'Your cuts follow the vital lines of the corpse')
```

Comment:
I was also thinking it might be useful to build a hash map of rituals to messages because it is not exactly clear what rituals are being checked here.